### PR TITLE
trie: better error reporting

### PR DESCRIPTION
**Pull request by holiman**
We get a lot of reports of "Unhandled trie error", which can be output from several different places. This PR makes it so we can determine which location causes the problem(s).

### CODE REVIEW

File: trie/secure_trie.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (t *StateTrie) TryGetAccount(key []byte) (*types.StateAccount, error) {
 	var ret types.StateAccount
 	res, err := t.trie.TryGet(t.hashKey(key))
 	if err != nil {
-		log.Error(fmt.Sprintf("Unhandled trie error: %v", err))
+		log.Error("Unhandled trie error in StateTrie.TGA", "err", err)
```
Comment (holiman) [inline trie/secure_trie.go:99]: In this location, we pass the `err` on to the caller, so it's IMO not correct to call it unhandled.

File: trie/secure_trie.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func (t *StateTrie) TryGetAccountWithPreHashedKey(key []byte) (*types.StateAccou
 	var ret types.StateAccount
 	res, err := t.trie.TryGet(key)
 	if err != nil {
-		log.Error(fmt.Sprintf("Unhandled trie error: %v", err))
+		log.Error("Unhandled trie error in StateTrie.TGAWPHK", "err", err)
```
Comment (holiman) [inline trie/secure_trie.go:116]: Same here, this is not unhandled, we pass it on

File: trie/proof.go [hunk 3]
```diff
@@ -63,7 +63,7 @@ func (t *Trie) Prove(key []byte, fromLevel uint, proofDb ethdb.KeyValueWriter) e
 			var err error
 			tn, err = t.resolveHash(n, prefix)
 			if err != nil {
-				log.Error(fmt.Sprintf("Unhandled trie error: %v", err))
+				log.Error("Unhandled trie error in Trie.Prove", "err", err)
```
Comment (holiman) [inline trie/proof.go:66]: This is not unhandled, it's passed to the caller

File: trie/stacktrie.go [hunk 4]
```diff
@@ -207,7 +206,7 @@ func (st *StackTrie) TryUpdate(key, value []byte) error {
 
 func (st *StackTrie) Update(key, value []byte) {
 	if err := st.TryUpdate(key, value); err != nil {
-		log.Error(fmt.Sprintf("Unhandled trie error: %v", err))
+		log.Error("Unhandled trie error in StackTrie.Update", "err", err)
```
Comment (holiman) [inline trie/stacktrie.go:209]: This can never happen, btw, so we could just nuke this occurrence.

File: trie/proof.go (hunk 3 above)
Comment (rjl493456442) [reply trie/proof.go:66]: should we nuke out this log?

**Review by rjl493456442 — APPROVED** [review]
LGTM, it's a really useful change.

File: trie/proof.go (hunk 3 above)
Comment (holiman) [reply trie/proof.go:66]: Let's leave it for now, to see if it's hit

**Pull request merged**